src/fsharp/TypeChecker.fs

```diff
@@ -5325,6 +5325,15 @@ and RecordNameAndTypeResolutions_IdeallyWithoutHavingOtherEffects_Delayed cenv e
         | _ -> ()
     dummyCheckedDelayed delayed 
 
+// Calls UnifyTypes, records name resolutions on failure, so that type information
+// can be collected for IntelliSense.
+and UnifyTypes_RecordNameResolutionsOnError cenv env m expectedTy actualTy =
```

Comment:
I think this would be called ``UnifyTypesAndRecover``.  Also it feels like you should be calling ``errorRecovery`` in the error path rather than ignoring any exception completely.